src/error.rs

```diff
@@ -71,3 +156,218 @@ impl ErrorCode {
 
     pub(crate) const NOT_INITIALIZED: ErrorCode = ErrorCode(0x8004_01F0);
 }
+
+#[repr(transparent)]
+struct BString {
+    ptr: RawPtr,
+}
+
+impl BString {
+    pub fn new() -> BString {
+        Self {
+            ptr: std::ptr::null_mut(),
+        }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.ptr.is_null() || self.len() == 0
+    }
+
+    pub fn len(&self) -> usize {
+        unsafe { SysStringLen(self.ptr) as usize }
```

Comment:
I'm assuming that `SysStringLen` can support null pointers? It would be also be good for us to get in the habit of noting why code marked as unsafe is in fact safe.

Reply:
[SysStringLen](https://docs.microsoft.com/en-us/windows/win32/api/oleauto/nf-oleauto-sysstringlen) can indeed be safely called with a null pointer (*"If `bstr` is null the return value is zero."*).